### PR TITLE
Update extensions to Cocoa Foundation classes.

### PR DESCRIPTION
**Pull request by bgulanowski**
Standardize category names and unify to have one category per class. Move files to "Additions" directory.

First step in updating to 2.0 model definitions.

cc @davidmuzi @dbart01 @gabrieloc

### CODE REVIEW

**Comment by davidmuzi** [comment]
We should have more unit tests to get some coverage for these classes.

mobile_buy_sdk_advanced_sample_xcodeproj

**Comment by bgulanowski** [comment]
@dbart01 if you have a chance you can help with this by copying over the tests I missed?

**Comment by bgulanowski** [comment]
Just realized that a bunch of the tests are in a later commit on these branches. Might be able to cherry pick.

**Comment by bgulanowski** [comment]
No, I was wrong. I was looking at the next PR, and it showed the same additions category files, but the tests are not there.

**Comment by bgulanowski** [comment]
@davidmuzi tests added.

**Comment by bgulanowski** [comment]
This has been rebased with additional changes.

File: Mobile Buy SDK/Mobile Buy SDK/Additions/NSString+BUYAdditions.h [hunk 1]
```diff
@@ -0,0 +1,129 @@
+//
+//  NSString+BUYAdditions.h
```
Comment (davidmuzi) [inline Mobile Buy SDK/Mobile Buy SDK/Additions/NSString+BUYAdditions.h]: It appears that the only 2 methods required are `buy_stringByAppendingBaseFileNameSuffix` and `buy_trim`

Comment (bgulanowski) [inline Mobile Buy SDK/Mobile Buy SDK/Additions/NSString+BUYAdditions.h]: …so far. There's a lot more code coming which relies on the additional code.

**Comment by davidmuzi** [comment]
LGTM 🎩

**Pull request merged**